packages/shared/src/settings/background.ts

```diff
@@ -10,24 +10,31 @@ import { BACKGROUND_COLOR_DEFAULT_VALUE } from './defaultValues';
  * @param options.id Custom suffix for the setting ids
  * @param options.defaultValue Default value for the background switch
  * @param options.defaultColor Default value for the background color
- * @returns {SettingBlock} Returns border settings
+ * @param options.hasDefaultColor Whether the background color should have a default value
+ * @param options.switchLabel Label for the background switch
+ * @returns {SettingBlock} Returns background settings
  */
 
 type BackgroundSettingsType = {
     id?: string;
     defaultValue?: boolean;
     defaultColor?: Color;
+    hasDefaultColor?: boolean;
+    switchLabel?: string;
 };
 
 export const getBackgroundSettings = (options?: BackgroundSettingsType): SettingBlock => {
+    const showDefaultColor = options?.hasDefaultColor !== undefined ? options?.hasDefaultColor : true;
     const hasId = options?.id ? `hasBackground${options.id}` : 'hasBackground';
     const colorId = options?.id ? `backgroundColor${options.id}` : 'backgroundColor';
-    const defaultColor = options?.defaultColor || BACKGROUND_COLOR_DEFAULT_VALUE;
+    const defaultColor = showDefaultColor ? options?.defaultColor || BACKGROUND_COLOR_DEFAULT_VALUE : undefined;
```

Comment:
```suggestion
    const defaultColor = !!options?.hasDefaultColor ? options?.defaultColor || BACKGROUND_COLOR_DEFAULT_VALUE : undefined;
```

Reply:
unfortunately not possible as hasDefaultColor is a boolean and if it is `false` we don't want to have a background color. if it is true or undefined we want to have the default color.